mttl/models/expert_model.py

```diff
@@ -21,10 +26,28 @@
 from mttl.models.expert_config import BaseExpertModelConfig
 from mttl.models.library.expert import Expert, ExpertInfo
 from mttl.models.library.expert_library import ExpertLibrary
-from mttl.models.modifiers.base import AutoModifierConfig
+from mttl.models.modifiers.base import (
+    AutoModifierConfig,
+    MergeableModifierMixin,
+    Modifier,
+)
 from mttl.models.modifiers.modify_model import modify_transformer
 
 
+@contextlib.contextmanager
+def disable_adapters(model):
+    """Context manager to disable all adapters in a model.
+
+    Args:
+        model (BaseExpertModel): The model to disable adapters in.
+    """
+    model.disable_adapters()
+
+    yield
+
+    model.enable_adapters()
```

Comment:
we should revert to the original state of the model.

Reply:
what do you mean?